firebase-appdistribution/src/main/java/com/google/firebase/appdistribution/impl/UpdateTaskCache.java

```diff
@@ -0,0 +1,47 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.appdistribution.impl;
+
+import com.google.android.gms.tasks.Task;
+
+/**
+ * A cache for Tasks, for use in cases where we only ever want one active task at a time for a
+ * particular operation.
+ */
+class TaskCache<T extends Task> {
+
+  /** A functional interface for a producer of a new Task. */
+  interface TaskProducer<T extends Task> {
```

Comment:
If you keep this, add `@FunctionalInterface`

Reply:
Done